### PR TITLE
Support Virtual Machines on Cycle

### PR DESCRIPTION
**Pull request by mattoni**
This PR begins the work to support virtual machines on Cycle. Details will come once the feature has been fully tested and released.

### CODE REVIEW

File: components/schemas/containers/summaries/ImageSummary.yml [hunk 1]
```diff
@@ -19,9 +19,13 @@ properties:
...
+        enum:
+          - "virtual-machine-v1"
+      id:
+        $ref: ../../ID.yml
   service:
     anyOf:
       - $ref: ../ServiceContainerIdentifier.yml
       - type: "null"
-    description: If a service container, the identifier specifying which service it is.
+    description: If a service container, the identifier specifying which service it is
```
Comment (blewisCycle) [inline components/schemas/containers/summaries/ImageSummary.yml:31]: All descriptions require punctuation.

File: components/schemas/vms/VirtualMachine.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    type: boolean
+    description: A boolean flag indicating if this virtual machine is deprecated.
+  lock:
+    type: boolean
+    description: Prevents this virtual machine from being deleted when true.
+  meta:
+    anyOf:
+      - $ref: ./VirtualMachineMeta.yml
+      - type: "null"
+required:
```
Comment (blewisCycle) [inline components/schemas/vms/VirtualMachine.yml:71]: And reason this is put after the properties and not before like we do everywhere else?  I know this is valid but seems we'd want to remain consistent.

File: components/schemas/vms/VirtualMachineMeta.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+title: VirtualMachineMeta
+type: object
+description: Fields that can be included on a per-virtual machine basis, specific to that virtual machine.
+properties:
+  domains:
+    anyOf:
+      - type: array
+        items:
+          type: object
+          description: Any associated LINKED records for this virtual machine, and their fully-qualified domain name (fqdn)
```
Comment (blewisCycle) [inline components/schemas/vms/VirtualMachineMeta.yml:10]: punctuation

File: components/schemas/vms/VirtualMachineVolume.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+properties:
+  id:
+    $ref: ../ID.yml
+  hash:
+    type: string
+    description: A unique hash identifying the volume for the virtual machine.
+  config:
+    $ref: ./config/volumes/VirtualMachineVolumeConfig.yml
+    description: The configuration details for the virtual machine's volume.
+required:
```
Comment (blewisCycle) [inline components/schemas/vms/VirtualMachineVolume.yml:14]: Same question re: required after properties

File: components/schemas/vms/config/VirtualMachineConfig.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+title: VirtualMachineConfig
+type: object
+description: |
+  Defines the network and deployment configurations for a virtual machine.
+properties:
+  network:
+    $ref: network/VirtualMachineNetworkConfig.yml
+  deploy:
+    $ref: ./deploy/VirtualMachineDeployConfig.yml
+required:
```
Comment (blewisCycle) [inline components/schemas/vms/config/VirtualMachineConfig.yml:10]: Same Question

File: components/schemas/vms/images/VirtualMachineBaseImage.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    description: The official homepage for the base image.
+  versions:
+    type: array
+    description: A list of available versions for the base image, including details like version number and disk image URL.
+    items:
+      type: object
+      properties:
+        identifier:
+          type: string
+          description: A unique identifier for the version (e.g., "alpine-3-19").
```
Comment (blewisCycle) [inline components/schemas/vms/images/VirtualMachineBaseImage.yml:24]: Use an example property

File: components/schemas/vms/images/VirtualMachineBaseImage.yml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        version:
+          type: string
+          description: The version number of the base image (e.g., "3.19").
```
Comment (blewisCycle) [inline components/schemas/vms/images/VirtualMachineBaseImage.yml:27]: Example property

File: platform/api.yml [hunk 8]
```diff
@@ -368,10 +374,10 @@ paths:
     $ref: "./paths/containers/instances/tasks.yml"
   "/v1/containers/{containerId}/instances/{instanceId}/volumes":
     $ref: "./paths/containers/instances/volumes.yml"
-  "/v1/containers/{containerId}/instances/{instanceId}/telemetry/resources/report":
-    $ref: paths/containers/instances/telemetry/report.yml
-  "/v1/containers/{containerId}/instances/{instanceId}/telemetry/resources/stream":
-    $ref: paths/containers/instances/telemetry/stream.yml
+  ? "/v1/containers/{containerId}/instances/{instanceId}/telemetry/resources/report"
```
Comment (blewisCycle) [inline platform/api.yml:377]: Is this intentional?

Comment (mattoni) [reply platform/api.yml:377]: my yaml formatter does this.

**Review by blewisCycle — APPROVED** [review]

**Pull request merged**